e2e/stack.go

```diff
@@ -53,89 +51,139 @@ func New(
 	monomerEngineURL,
 	monomerCometURL,
 	opNodeURL *url.URL,
-	contractsRootDir string,
-	l1BlockTime time.Duration,
+	deployConfigDir string,
+	l1stateDumpDir string,
+	l1BlockTime uint64,
 	eventListener EventListener,
 ) *Stack {
 	return &Stack{
-		anvilURL:         anvilURL,
 		monomerEngineURL: monomerEngineURL,
 		monomerCometURL:  monomerCometURL,
 		opNodeURL:        opNodeURL,
-		contractsRootDir: contractsRootDir,
+		deployConfigDir:  deployConfigDir,
+		l1stateDumpDir:   l1stateDumpDir,
 		eventListener:    eventListener,
 		l1BlockTime:      l1BlockTime,
 	}
 }
 
+// helper shim to allow decoding of hex nonces
+type auxDumpAccount struct {
+	Balance     string                 `json:"balance"`
+	Nonce       string                 `json:"nonce"`
```

Comment:
That's pretty annoying that the `allocs-l1.json` file has the nonce as a hex value. For my own understanding is there any reasoning behind why they store it this way in that file or is it just for backwards compatibility? I was thinking about having an additional step in `make setup-e2e` to do this conversion but I feel like that's probably unnecessary for such a small change

Reply:
Yeah I agree with you, I think that having the separate script in the makefile would get very messy and I like your approach of doing the conversion with the `aux` account. Thanks for the thorough explanation!